test/new-e2e/runner/parameters/store_config_file.go

```diff
@@ -8,55 +8,97 @@ package parameters
 import (
 	"os"
 
+	"encoding/json"
+
 	"gopkg.in/yaml.v2"
 )
 
+// Configuration file example
+// configParams:
+//   aws:
+//     keyPairName: "totoro"
+//     publicKeyPath: "/home/totoro/.ssh/id_rsa.pub"
+//   agent:
+//     apiKey: "00000000000000000000000000000000"=
+// stackParams:
+//   ddinfra:
+//     aws/someVariable: "ponyo"
+
 type Config struct {
-	StackParams StackParams `yaml:"stackParams"`
+	ConfigParams ConfigParams                 `yaml:"configParams"`
+	StackParams  map[string]map[string]string `yaml:"stackParams"`
 }
 
-type StackParams struct {
-	DDInfra DDInfra `yaml:"ddinfra"`
+type ConfigParams struct {
+	AWS   AWS   `yaml:"aws"`
+	Agent Agent `yaml:"agent"`
 }
-
-type DDInfra struct {
-	AWS AWS `yaml:"aws"`
+type AWS struct {
+	KeyPairName   string `yaml:"keyPairName"`
+	PublicKeyPath string `yaml:"publicKeyPath"`
 }
 
-type AWS struct {
-	DefaultKeyPairName string `yaml:"defaultKeyPairName"`
+type Agent struct {
+	APIKey string `yaml:"apiKey"`
+	APPKey string `yaml:"appKey"`
 }
 
+var _ valueStore = &configFileValueStore{}
+
 type configFileValueStore struct {
-	values map[string]string
+	config          Config
+	stackParamsJson string
 }
 
 func NewConfigFileValueStore(path string) (configFileValueStore, error) {
+	store := configFileValueStore{}
 	content, err := os.ReadFile(path)
 	if err != nil {
-		return configFileValueStore{}, err
+		return store, err
 	}
 
-	config := Config{}
-	err = yaml.Unmarshal(content, &config)
+	err = yaml.Unmarshal(content, &store.config)
+
 	if err != nil {
-		return configFileValueStore{}, err
+		return store, err
 	}
 
-	valueStore := configFileValueStore{values: make(map[string]string)}
-
-	defaultKeyPairName := config.StackParams.DDInfra.AWS.DefaultKeyPairName
-	if defaultKeyPairName != "" {
-		valueStore.values[GetDefaultKeyPairParamName()] = defaultKeyPairName
+	// parse StackParams to json string
+	stackParams := map[string]string{}
+	for namespace, submap := range store.config.StackParams {
+		for key, value := range submap {
+			stackParams[namespace+":"+key] = value
+		}
 	}
-	return valueStore, nil
+	b, err := json.Marshal(stackParams)
+	if err != nil {
+		return store, err
+	}
+	store.stackParamsJson = string(b)
+
+	return store, err
 }
 
 // Get returns parameter value.
-func (s configFileValueStore) get(key string) (string, error) {
-	value, ok := s.values[key]
-	if ok {
-		return value, nil
+func (s configFileValueStore) get(key StoreKey) (string, error) {
+	var value string
+
+	switch key {
+	case APIKey:
+		value = s.config.ConfigParams.Agent.APIKey
+	case APPKey:
+		value = s.config.ConfigParams.Agent.APPKey
+	case KeyPairName:
+		value = s.config.ConfigParams.AWS.KeyPairName
+	case PublicKeyPath:
```

Comment:
`PublicKeyPath` and `StackParameters` seems to not be injected in the configmap.

Reply:
Added `PublicKeyPath`, `StackParameters` is injected at https://github.com/DataDog/datadog-agent/blob/6167c4ab46c6ff962c3e98a4f120e32c9b83b1ce/test/new-e2e/runner/configmap.go#L99-L112